code/__HELPERS/unsorted.dm

```diff
@@ -1035,19 +1033,29 @@ proc/DuplicateObject(obj/original, var/perfectcopy = 0 , var/sameloc = 0)
 				var/datum/coords/C_trg = refined_trg[B]
 				if(C_src.x_pos == C_trg.x_pos && C_src.y_pos == C_trg.y_pos)
 
+					var/old_name = T.name
 					var/old_dir1 = T.dir
 					var/old_icon_state1 = T.icon_state
 					var/old_icon1 = T.icon
 
+					var/datum/gas_mixture/new_air
+					if(T.air)
+						new_air = new
+						new_air.copy_from(T.air)
+
 					if(platingRequired)
 						if(istype(B, /turf/space))
 							continue moving
 
 					var/turf/X = B.ChangeTurf(T.type)
+					X.name = old_name
 					X.dir = old_dir1
 					X.icon_state = old_icon_state1
 					X.icon = old_icon1 //Shuttle floors are in shuttle.dmi while the defaults are floors.dmi
 
+					if(new_air)
+						X.air = new_air
```

Comment:
I think this works but it feels wrong to me for some reason. Can't think of anything better though.